go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -229,53 +228,23 @@ func (oc *Controller) addEgressFirewall(egressFirewall *egressfirewallapi.Egress
 	}
 
 	ipv4HashedAS, ipv6HashedAS := nsInfo.addressSet.GetASHashNames()
-	err = oc.addEgressFirewallRules(ipv4HashedAS, ipv6HashedAS, egressFirewall.Namespace, egressFirewallStartPriorityInt)
+	err = oc.addEgressFirewallRules(ipv4HashedAS, ipv6HashedAS, egressFirewall.Namespace, egressFirewallStartPriorityInt, txn)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (oc *Controller) updateEgressFirewall(oldEgressFirewall, newEgressFirewall *egressfirewallapi.EgressFirewall) error {
-	// block all external traffic in this namespace
-	nsInfo, err := oc.waitForNamespaceLocked(newEgressFirewall.Namespace)
-	if err != nil {
-		return fmt.Errorf("cannot update egressfirewall in %s:%v", newEgressFirewall.Namespace, err)
-	}
-	addressSet := nsInfo.addressSet
-	nsInfo.Unlock()
-	priority, err := strconv.Atoi(types.EgressFirewallStartPriority)
-	if err != nil {
-		return fmt.Errorf("cannot update egressfirewall in %s:%v", newEgressFirewall.Namespace, err)
-	}
-
-	ipv4ASHashName, ipv6ASHashName := addressSet.GetASHashNames()
-	match := generateMatch(ipv4ASHashName, ipv6ASHashName, []matchTarget{
-		{matchKindV4CIDR, "0.0.0.0/0"},
-		{matchKindV6CIDR, "::/0"},
-	},
-		[]egressfirewallapi.EgressFirewallPort{},
-	)
-
-	err = oc.createEgressFirewallRules(priority, match, "drop", newEgressFirewall.Namespace+"-blockAll")
-	if err != nil {
-		return fmt.Errorf("cannot update egressfirewall in %s:%v", newEgressFirewall.Namespace, err)
-	}
-
-	updateErrors := oc.deleteEgressFirewall(oldEgressFirewall)
+func (oc *Controller) updateEgressFirewall(oldEgressFirewall, newEgressFirewall *egressfirewallapi.EgressFirewall, txn *util.NBTxn) error {
+	updateErrors := oc.deleteEgressFirewall(oldEgressFirewall, txn)
```

Comment:
@JacobTanenbaum I'm probably missing something, but what now calls createEgressFirewallRules() for the update case, if anything?

Reply:
@JacobTanenbaum ah, nevermind, it's the addEgressFirewall() bit just below.

Instead of putting that into the errors.Wrapf(), could you take it out and do something like:

```
if err := oc.addEgressFirewall(newEgressFirewall, txn); err != nil {
    updateErrors = errors.Wrapf(updateErrors, err)
}
```

Reply:
done